File: app/src/main/java/com/ed/redditapp/ui/imageview/ImageViewActivity.java
package com.ed.redditapp.ui.imageview;

import android.graphics.Bitmap;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.ed.redditapp.databinding.ActivityImageviewBinding;

import javax.inject.Inject;

public class ImageViewActivity extends AppCompatActivity {
    public static final String EXTRA_IMAGE_URL = "extra_image_url";

    private ActivityImageviewBinding binding;

    @Inject ImageViewPresenter presenter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityImageviewBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        DaggerImageViewDaggerComponent
                .builder()
                .imageViewDaggerModule(new ImageViewDaggerModule(this))
                .build()
                .inject(this);

        presenter.onLoaded(getIntent().getStringExtra(EXTRA_IMAGE_URL));
    }

    public void setImage(Bitmap image) {
        binding.imageview.setImageBitmap(image);
    }
}

File: app/src/main/java/com/ed/redditapp/ui/imageview/ImageViewActivity.kt
package com.ed.redditapp.ui.imageview

import android.net.Uri
import android.os.Bundle
import com.ed.redditapp.BaseActivity
import com.ed.redditapp.databinding.ActivityImageviewBinding
import javax.inject.Inject

class ImageViewActivity: BaseActivity<ActivityImageviewBinding>() {
    companion object {
        val EXTRA_IMAGE_URL = "extra_image_url"
    }

    override val binding = ActivityImageviewBinding.inflate(layoutInflater)

    @Inject lateinit var presenter: ImageViewPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        DaggerImageViewDaggerComponent
                .builder()
                .imageViewDaggerModule(ImageViewDaggerModule(this))
                .build()
                .inject(this)

        presenter.onLoaded(intent.getStringExtra(EXTRA_IMAGE_URL)!!)
    }

    fun setImage(url: String) {
        binding.imageview.showImage(Uri.parse(url))
    }
}